### ISSUE TITLE
Request: Make typing not suck (line-by-line)

### ISSUE THREAD

**Issue by cwbit** [PC-Logix/SimpleInfoPanels#6]
Adv Info Panel's editor is a line-by-line editor which sucks because it's basically like using an old Brother Word Processor; if you make a mistake, or think of something else to add, it's basically faster to scrap the "text card" and start over. You have to clear and edit each line of the display individually.

It's important that line spacing be preserved (e.g. i need this text on one line and this text two lines down from that) but if the text editor was more like an actual rich text editor (where hitting enter would move the lines down, etc) that'd be ideal.

**Comment by CaitlynMainer** [comment]
The Text Card editor is planned to be a Text Area, supporting enter for newlines, also considering something BBCode-esque for formatting/color, thoughts?

**Comment by cwbit** [comment]
Honestly NOT requiring a 'text card' (since it's the most common usage of the panel) would be ideal.

TLDR; The trick is to make it really fast to use. 

Like for tutorials we need to be able to hammer them out all over the place with little to no setup. Having to spawn in text cards is a slow-down. Having to specify a color with text would be a slowdown. For colors, the Adv Info Panel setup is nice where u just need to click a box to change the color (except dark grey should be default since black is too dark, and green is ugly as sin)

It would be nice to somehow support more advanced text tho for things like computer-craft controlled monitors - eg a power status card that could show low power settings in RED or a computer monitored farm that would change text color depending on status (ON|OFF) or inventory level (HIGH|LOW|MID) etc

Not sure how difficult something like MARKDOWN support would be (I think a C-parser already exists)

**Comment by cwbit** [comment]
oh but also yeah textarea w/ newline support is +1

**Comment by CaitlynMainer** [comment]
Well, background color would be selected on the panel as a whole, text color/formatting I was considering something like bbcode so you could do per word, (or even per letter) coloring, markdown seems like it might be a bit overkill to me.  OC/CC Control is planned, but will likely not be in the initial release.  The text card was planned so you could move panels around if you needed, without having to get hacky and store the text in an Item Block when broken.  I'm open to ideas though.

**Comment by cwbit** [comment]
For us, text always gets colored as a whole, too. Like we do dark-grey background with cyan text or orange text, etc where all the text is one color. If you could do set all-text color and then override locally word or letter that'd be best of both worlds.

For text cards, Any way to store the text like CoFH stores block configs when wrenched? So no card needed and just wrench if need to move. Would also keep any other network/power/mod cards, too.

Or, for text, make it use cards only if you need to copy/paste. Put a card in, save text to card, move to other panel, insert card, paste from card. So that it's like opt-in

**Comment by CaitlynMainer** [comment]
Wrenching stuffs the block config into the NBT of an Item Block, then restores it on placing, exactly what I'm wanting to avoid, I did it on OpenFM's radio block, and it's a PITA at times.  I'll look into it though.
